runtime/lua/vim/lsp.lua

```diff
@@ -1042,6 +1042,7 @@ do
     if tbl_isempty(all_buffer_active_clients[bufnr] or {}) then
       return
     end
+    lsp.diagnostic.restore_extmarks(bufnr)
```

Comment:
Given that we don't use any of the additional information that is being provided in these handlers, wouldn't it make sense to try to encapsulate this within the diagnostic module with a dedicated `buf_attach` `on_lines` listener?

I think it is kinda nice that currently `lsp.diagnostic` is pretty self-contained. 
We also wouldn't have to expose `M.save_extmarks` and `M.restore_extmarks` and they currently look a lot like implementation detail and not like something we'd want users to call directly?

Reply:
makes sense. I'll change it

Reply:
done :)